ledger-core/v2/virtual/integration/vdelegated_test.go

```diff
@@ -89,3 +89,91 @@ func TestVirtual_VDelegatedCallRequest(t *testing.T) {
 		require.Failf(t, "", "timeout")
 	}
 }
+
+func TestVirtual_VDelegatedCallRequest_GetBalance(t *testing.T) {
+	t.Log("C4982")
+
+	server, ctx := utils.NewServer(nil, t)
+	defer server.Stop()
+
+	testBalance := uint32(555)
+	objectRef := gen.UniqueReference()
+
+	{
+		// send VStateReport: save wallet
+		stateID := gen.UniqueIDWithPulse(server.GetPulse().PulseNumber)
+		rawWalletState := makeRawWalletState(t, testBalance)
+		payloadMeta := &payload.VStateReport{
+			Callee:                        objectRef,
+			UnorderedPendingCount:         1,
+			UnorderedPendingEarliestPulse: pulse.OfNow(),
+			ProvidedContent: &payload.VStateReport_ProvidedContentBody{
+				LatestDirtyState: &payload.ObjectState{
+					Reference: stateID,
+					Class:     testwalletProxy.GetClass(),
+					State:     rawWalletState,
+				},
+			},
+		}
+		msg := utils.NewRequestWrapper(server.GetPulse().PulseNumber, payloadMeta).Finalize()
+		server.SendMessage(ctx, msg)
```

Comment:
use 
```
	server.WaitIdleConveyor()
	server.ResetActiveConveyorFlag()
```
to ensure messages order before/after pulse

Reply:
done